src/promql/src/planner.rs

```diff
@@ -82,6 +84,24 @@ const FIELD_COLUMN_MATCHER: &str = "__field__";
 /// Special modifier for cross schema query
 const SCHEMA_COLUMN_MATCHER: &str = "__schema__";
 
+/// A evaluated literal in PromQL
+enum LiteralExpr {
+    /// `float` or `string` literal in PromQL. (e.g. `1.2`, `"host"`)
+    Const(DfExpr),
+    /// `scalar` function in PromQL, we only know the literal value after plan executed. (e.g. `scalar(host)`)
+    Runtime(LogicalPlan),
```

Comment:
Why treat `scalar` as literal?

Reply:
<img width="588" alt="image" src="https://github.com/GreptimeTeam/greptimedb/assets/46434815/705f5063-b13d-4013-afc3-e58f326dfcc5">


Because the definition of scalar function will return a scalar，so I treat it as a runtime literal